calyx-opt/src/passes/top_down_compile_control.rs

```diff
@@ -828,6 +834,7 @@ impl Schedule<'_, '_> {
             &if_stmt.tbranch,
             tru_transitions,
             early_transitions,
+            false,
```

Comment:
Why does this unconditionally pass `false` instead of threading through `early_transitions`? Is it because we want to enable this if we see it in a `seq` or something?

Reply:
We only want to enable `@fast` on the parent control, I believe, unless you had different semantics in mind?

Reply:
Okay, that makes sense! We should eventually get rid of the other `early-transition` logic entirely in a follow-up PR.

Reply:
Hmmm, there may be more cases in which we can apply `early-transition`, for example, if (as @sampsyo said) one could prove that two adjacent groups cannot be `done` at the same time; here, we could reuse `@fast`'s functionality, but there might be another case where it does apply recursively. That can be dealt with later though, it's not too hard to just re-add a parameter to all the functions.